activerecord/lib/active_record/relation.rb

```diff
@@ -670,7 +670,18 @@ def values
     end
 
     def inspect
-      entries = records.take([limit_value, 11].compact.min).map!(&:inspect)
+      num_to_take = [limit_value, 11].compact.min
+      # A relation may be initialized with values that were not loaded from the
+      # database. Therefore, if there are values, they must be preserved by not
+      # loading from the database; else, use a new relation with a limit, to
+      # avoid loading an unbounded number of records into memory unnecessarily.
+      entries =
+        if values.any?
```

Comment:
shouldn't this be `@records.present?` instead. So if the `@records` are loaded dont yield the query, otherwise do a limit query?